---
title: "multiple ENU variant effect prediction analysis - snv"
output: html_document
date: "2023-10-12"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Preparation

## Load libraries
```{r}
library(ggplot2)
library(stringr)
library(dplyr)
library(ggpubr)
library(purrr)
library(rstatix)
library(emmeans)
library(viridis)
```

## Functions
```{r}
# sum similar variants
sum_and_replace_rows <- function(data, strings_to_match) {
  # Initialize an empty dataframe with the same columns as the input data
  result_data <- data.frame()
  
  # Loop through each string to match
  for (string_to_match in strings_to_match) {
    # Define the pattern to match
    pattern <- paste0("^", string_to_match)
    
    # Filter and group the data
    grouped_data <- data %>%
      filter(grepl(pattern, variant_type)) %>%
      group_by(sample, condition, cycle) %>%
      summarise(obs.mut = sum(obs.mut), .groups = 'drop')
    
    # Add the matched variant_type to the grouped data
    grouped_data <- grouped_data %>%
      mutate(variant_type = string_to_match)
    
    # Append the grouped data to the result_data
    result_data <- bind_rows(result_data, grouped_data)
    
    # Remove the matched rows from the original data
    data <- data %>%
      filter(!grepl(pattern, variant_type))
  }
  
  # Bind the remaining original data to the result_data
  result_data <- bind_rows(result_data, data)
  
  return(result_data)
}

# combine the permutation pvalues from each sample
# to assess if group of samples is significantly different from expected
combine_pvalues_fisher <- function(p_values) {
    chi_sq_stat = -2 * sum(log(p_values))
    degrees_of_freedom = 2 * length(p_values)
    p_combined = pchisq(chi_sq_stat, degrees_of_freedom, lower.tail = FALSE)
    return(p_combined)
}

# function to calculate significance with results coming from
# observed and downsampled
ratio_significance <- function(ratios, group_vars) {
  
  res <- ratios %>%
    group_by_at(vars(one_of(group_vars))) %>%
    summarise(pvalue = combine_pvalues_fisher(pvalue)) %>%
    mutate(pvalue_symbol = case_when(
      pvalue <= 0.0001 ~ '****',
      pvalue <= 0.001 ~ '***',
      pvalue <= 0.01 ~ '**',
      pvalue <= 0.05 ~ '*',
      TRUE ~ NA))
}

# function to calculate significance between condition groups
group_significance <- function(combined_regions, group_vars, condition, group_levels){
  formula <- as.formula(paste("obs.mut ~", condition))
  
  results <- combined_regions %>%
    group_by_at(vars(one_of(group_vars))) %>%
    summarise(
      p_value = wilcox.test(formula = formula, 
                            data = cur_data(),
                            exact = FALSE)$p.value
    ) %>%
    ungroup() %>%
    mutate(p_adjusted = p.adjust(p_value, method = "BH"),
           significant_p_adjusted = case_when(
                            p_adjusted <= 0.0001 ~ '****',
                            p_adjusted <= 0.001 ~ '***',
                            p_adjusted <= 0.01 ~ '**',
                            p_adjusted <= 0.05 ~ '*',
                            TRUE ~ NA_character_)
    )

  # Order by region
  results$variant_type <- factor(results$variant_type, levels = group_levels)
  
  return(results)
}

calculate_stat_summary <- function(observed, group_vars, mut_var){
  results <- observed %>%
    group_by_at(vars(one_of(group_vars))) %>%
    summarise(mean = mean(!!sym(mut_var), na.rm = TRUE),
              sd = sd(!!sym(mut_var), na.rm = TRUE))
}

theme_Publication <- function(base_size=16, base_family="helvetica") {
      library(grid)
      library(ggthemes)
      (theme_foundation(base_size=base_size, base_family = "")
       + theme(plot.title = element_text(face = "bold",
                                         size = rel(1.2), hjust = 0.5),
               text = element_text(),
               panel.background = element_rect(colour = NA),
               plot.background = element_rect(colour = NA),
               panel.border = element_rect(colour = NA),
               axis.title = element_text(face = "bold",size = rel(1)),
               axis.title.y = element_text(angle=90,vjust =2),
               axis.title.x = element_text(vjust = -0.2),
               axis.text = element_text(), 
               axis.line = element_line(colour="black"),
               axis.ticks = element_line(),
               panel.grid.major = element_line(colour="#f0f0f0"),
               panel.grid.minor = element_blank(),
               legend.key = element_rect(colour = NA),
               legend.position = "bottom",
               legend.direction = "horizontal",
               #legend.key.size= unit(0.2, "cm"),
               legend.margin = unit(0, "cm"),
               legend.title = element_text(face="italic"),
               plot.margin=unit(c(10,5,5,5),"mm"),
               strip.background=element_rect(colour="#f0f0f0",fill="#f0f0f0"),
               strip.text = element_text(face="bold")
          ))
      
}

scale_fill_Publication <- function(...){
      library(scales)
      discrete_scale("fill","Publication",manual_pal(values = c("#386cb0","#fdb462","#7fc97f","#ef3b2c","#662506","#a6cee3","#fb9a99","#984ea3","#ffff33")), ...)

}

scale_fill_Publication_2 <- function(...){
      library(scales)
      discrete_scale("fill","Publication",manual_pal(values = c("#7fc97f","#ef3b2c","#386cb0","#fdb462","#662506","#a6cee3","#fb9a99","#984ea3","#ffff33")), ...)

}

scale_colour_Publication <- function(...){
      library(scales)
      discrete_scale("colour","Publication",manual_pal(values = c("#386cb0","#fdb462","#7fc97f","#ef3b2c","#662506","#a6cee3","#fb9a99","#984ea3","#ffff33")), ...)

}

wrap_labels <- function(labels, width = 5) {
  wrapped_labels <- str_wrap(labels, width)
  names(wrapped_labels) <- names(labels)
  return(wrapped_labels)
}
```

## Load sample table
```{r}
sample_table <- read.table("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/1-data/sample_table_filtered.txt", header = TRUE, sep = "\t")
```

## Load ratio results
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/1-calculate-ratio/cells")

# consequences
consequences.main.mean <- read.csv("consequence_main_oe_mean.csv", header = TRUE, row.names = NULL)
# consequences.main.median <- read.csv("consequence_main_oe_mean.csv", header = TRUE, row.names = NULL)
# # consequences.sub.mean <- read.csv("observed_consequence_subtypes_ratio_mean.csv", header = TRUE, row.names = NULL)
# # consequences.sub.median <- read.csv("observed_consequence_subtypes_ratio_median.csv", header = TRUE, row.names = NULL)
consequences.syn_non.mean <- read.csv("consequence_synonymous_nonsynonymous_oe_mean.csv", header = TRUE, row.names = NULL)
# consequences.syn_non.median <- read.csv("observed_consequence_synonymous_nonsynonymous_ratio_median.csv", header = TRUE, row.names = NULL)

# alpha missense - score above 0.564 likely pathogenic. range 0-Inf
# use absolute
# am.mean <- read.csv("observed_alpha-missense_absolute_ratio_mean.csv", header = TRUE, row.names = NULL)
# am.median <- read.csv("observed_alpha-missense_absolute_ratio_median.csv", header = TRUE, row.names = NULL)
# am.sum <- read.csv("observed_alpha-missense_absolute_ratio_sum.csv", header = TRUE, row.names = NULL)

# cadd - observed variant has negative values while simulated variant has positive values - more positive more damaging
# need to redo and classify into observed and simulated and then count
# classify into insertion and deletion
# use non-absolute
cadd.mean <- read.csv("cadd_phred_oe_mean.csv", header = TRUE, row.names = NULL)

# gerp - score above 2 indicates constrained site. Range from -12 to 6
# need to redo and classify into constrained and non and then count
# also need to prevent NA in control when calculating ratio
# use non-absolute and mean
# gerp.mean <- read.csv("observed_gerp_non-absolute_ratio_mean.csv", header = TRUE, row.names = NULL)
# gerp.median <- read.csv("observed_gerp_non-absolute_ratio_median.csv", header = TRUE, row.names = NULL)

# transcription factor score change
# use absolute
# tf.mean <- read.csv("observed_tf_absolute_ratio_mean.csv", header = TRUE, row.names = NULL)
# tf.median <- read.csv("observed_tf_absolute_ratio_median.csv", header = TRUE, row.names = NULL)
# tf.sum <- read.csv("observed_tf_absolute_ratio_sum.csv", header = TRUE, row.names = NULL)

# phyloP - conserved sites are positive and non-conserved are negative
# need to redo and classify into conserved and non-conserved and then count
# use non-absolute
# phylop.mean <- read.csv("observed_phylop_non-absolute_ratio_mean.csv", header = TRUE, row.names = NULL)
# phylop.median <- read.csv("observed_phylop_non-absolute_ratio_median.csv", header = TRUE, row.names = NULL)

# phastcons - probability of negative selection from 0-1
# use non-absolute
# phastcons.mean <- read.csv("observed_phastCons_non-absolute_ratio_mean.csv", header = TRUE, row.names = NULL)
# phastcons.median <- read.csv("observed_phastCons_non-absolute_ratio_median.csv", header = TRUE, row.names = NULL)
# phastcons.sum <- read.csv("observed_phastCons_non-absolute_ratio_sum.csv", header = TRUE, row.names = NULL)
```

# Analysis

## Consequences

### Number of different types of variants
```{r}
consequences.main.mean <- merge(consequences.main.mean,
                                    sample_table,
                                    by = "sample")

print(consequences.main.mean %>%
  group_by(variant_type) %>%
  summarize(total = sum(obs.mut)),
  n = Inf)

# print(consequences.syn_non.mean %>%
#   group_by(variant_type) %>%
#   summarize(total = sum(obs.mut)),
#   n = Inf)
```

### Get number of SNV vs INDEL variants
```{r}
consequences.main.mean <- consequences.main.mean %>%
  mutate(mutation_type = case_when(
    variant_type == "inframe insertion" ~ "INDEL",
    variant_type == "inframe deletion" ~ "INDEL",
    variant_type == "frameshift variant" ~ "INDEL",
    .default = "SNV"
    )
  )

consequences.main.mean %>%
  group_by(mutation_type) %>%
  summarize(total = sum(obs.mut))
```

### Get number of coding vs non-coding
```{r}
consequences.main.mean <- consequences.main.mean %>%
  mutate(group = case_when(
    variant_type == "synonymous variant" ~ "coding",
    variant_type == "missense variant" ~ "coding",
    variant_type == "start lost" ~ "coding",
    variant_type == "stop gained" ~ "coding",
    variant_type == "stop lost" ~ "coding",
    variant_type == "splice acceptor variant" ~ "coding",
    variant_type == "splice donor variant" ~ "coding",
    variant_type == "inframe insertion" ~ "coding",
    variant_type == "inframe deletion" ~ "coding",
    variant_type == "frameshift variant" ~ "coding",
    variant_type == "coding sequence variant / NMD transcript variant" ~ "coding",
    variant_type == "synonymous variant / NMD transcript variant" ~ "coding",
    .default = "non-coding"
    )
  )

consequences.main.mean %>%
  group_by(group) %>%
  summarize(total = sum(obs.mut))
```

# Consequences

## By cycle

##### All Variants
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences")
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle")

consequences.main.mean$cycle <- factor(consequences.main.mean$cycle)

stats_summary <- consequences.main.mean %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE))

# observed
pdf("consequences_cycle_main_mean.pdf",
    width = 30,
    height = 30)
ggplot(consequences.main.mean, aes(x = condition, y = obs.mut, fill = cycle)) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, fill = cycle, y = mean, ymin = mean - sd, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    # geom_text(data = generic_regions_ratio_significance, 
    #           aes(x = region, y = Inf, label = pvalue_symbol), # pvalue of observed/expected
    #           vjust = 0.8, hjust = 1, position=position_dodge(width = 0.75), size = 7, angle = 90) +
    # geom_text(data = generic_regions_group_significance,
    #           aes(x = region, y = Inf, fill = NA, label = significant_p_adjusted), # pvalue of control vs enu
    #           vjust = 15, hjust = 0.5, size = 8, color = "firebrick") +
    labs(x = "Condition", y = "Observed Variants Per Cell") +
    facet_wrap(.~variant_type, scales = "free", labeller = labeller(variant_type = label_wrap_gen(30))) + 
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.justification = c("center","top"),
          legend.position = "top",
          axis.text.x = element_text(size = 12))
dev.off()
```

##### Synonymous and Non-Synonymous
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/synonymous_nonsynonymous")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/synonymous_nonsynonymous")

# set levels
consequences.syn_non.mean$cycle <- factor(consequences.syn_non.mean$cycle)
consequences.syn_non.mean$condition <- factor(consequences.syn_non.mean$condition)
consequences.syn_non.mean$variant_type <- factor(consequences.syn_non.mean$variant_type,
                                                 levels = c("synonymous", "nonsynonymous"))

# stats
for (variant in unique(consequences.syn_non.mean$variant_type)) {
  
  filtered_data <- consequences.syn_non.mean %>%
                         filter(variant_type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.mut ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.mut ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("consequences.syn_non.mean.pvalue", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- consequences.syn_non.mean %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE))

write.csv(stats_summary, "consequences.syn_non.mean.summary.csv")

# new facet labels
labels <- c("Synonymous",
             "Non-Synonymous")
names(labels) <- c("synonymous", "nonsynonymous")

# bar plot
pdf("consequences.syn_non.mean.pdf",
    width = 4.25,
    height = 4)
ggplot(consequences.syn_non.mean, aes(x = condition, y = obs.mut, fill = as.factor(cycle))) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
  
    geom_errorbar(data = stats_summary, aes(x = condition, fill = as.factor(cycle), y = mean, ymin = mean - sd, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", y = "Observed Variants Per Cell", fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12),
        strip.text = element_text(size = 12)) +
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~variant_type,
               scale = "fixed",
               nrow = 1,
               labeller = labeller(variant_type = labels))
dev.off()
```

##### Synonymous / Non-Synonymous
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/synonymous_nonsynonymous")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/synonymous_nonsynonymous")

consequences.syn_non.mean <- merge(consequences.syn_non.mean,
                                    sample_table,
                                    by = "sample")

# calculate N/ per cell
syn_non.mean.ratio <- consequences.syn_non.mean %>%
  select(sample, condition, cycle, variant_type, obs.mut) %>%
  spread(key = variant_type, value = obs.mut) %>%
  mutate(obs.mut_ratio = (nonsynonymous + 1) / (synonymous + 1))

# stats
syn_non.mean.ratio.pvalue <- summary(aov(obs.mut_ratio ~ condition * cycle, data = syn_non.mean.ratio))
write.csv(capture.output(syn_non.mean.ratio.pvalue), "syn_non.mean.ratio.pvalue.csv")

# get mean and sd for each group
stats_summary <- syn_non.mean.ratio %>%
  group_by(condition, cycle) %>%
  summarise(mean = mean(obs.mut_ratio, na.rm = TRUE),
            sd = sd(obs.mut_ratio, na.rm = TRUE))

write.csv(stats_summary, "syn_non.mean.ratio.summary.csv")

# box plot
pdf("syn_non.mean.ratio.pdf",
    width = 3,
    height = 4)
ggplot(syn_non.mean.ratio, aes(x = condition, y = obs.mut_ratio, fill = as.factor(cycle))) + 
    geom_hline(yintercept = 1, linetype = "dashed") +
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Observed Non-Syn. / Syn.\n Variants Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        axis.text.y = element_text(size = 10),
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12),
        strip.text = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU"))
dev.off()
```

### Specific SNV coding variants
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/snv_coding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/snv_coding")

# sum similar variants
main_variants <- c("synonymous variant",
                    "start retained variant",
                    "stop retained variant",
                    "stop gained",
                    "stop lost",
                    "start lost",
                    "inframe insertion",
                    "frameshift variant",
                    "inframe deletion",
                    "missense variant", 
                    "intron",
                    "splice",
                    "3 prime UTR variant",
                    "5 prime UTR variant",
                    "TF binding site variant",
                    "regulatory region",
                    "upstream gene variant",
                    "downstream gene variant",
                    "intergenic variant",
                    "non coding transcript exon variant",
                   "NMD transcript variant"
)

consequences.main.mean.snv.code <- sum_and_replace_rows(consequences.main.mean, main_variants)

# subset 
consequences_levels <- c("synonymous variant",
                         "stop retained variant",
                         "missense variant",
                         "start lost",
                         "stop gained",
                         "stop lost")
consequences.main.mean.snv.code <- subset(consequences.main.mean.snv.code, variant_type %in% consequences_levels)

# set levels
consequences.main.mean.snv.code$variant_type <- factor(consequences.main.mean.snv.code$variant_type, 
                                                       levels = consequences_levels)

consequences.main.mean.snv.code$cycle <- factor(consequences.main.mean.snv.code$cycle, 
                                                levels = c(1,3,6,9))

consequences.main.mean.snv.code$condition <- factor(consequences.main.mean.snv.code$condition, 
                                                    levels = c("control", "enu"))


# stats
for (variant in unique(consequences.main.mean.snv.code$variant_type)) {
  
  filtered_data <- consequences.main.mean.snv.code %>%
                         filter(variant_type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.mut ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.mut ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("consequences.main.mean.snv.code.pvalue.", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- consequences.main.mean.snv.code %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE))

write.csv(stats_summary, "consequences.main.mean.snv.code.summary.csv")

# new facet labels
labels <- c("Synonymous",
            "Stop retained",
            "Missense",
             "Start lost",
             "Stop gained",
             "Stop lost")
names(labels) <- consequences_levels

# bar plot
pdf("consequences.main.mean.snv.code.pdf",
    width = 10,
    height = 4)
ggplot(consequences.main.mean.snv.code, aes(x = condition, y = obs.mut, fill = cycle)) + 
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, fill = cycle, y = mean, ymin = mean - sd, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", y = "Observed Variants Per Cell", fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~variant_type,
               scale = "fixed",
               nrow = 1,
               labeller = labeller(variant_type = labels))
dev.off()
```

#### Non-coding variants
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/snv_noncoding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/snv_noncoding")

# subset 
consequences_levels <- c("intron",
                         "intergenic variant",
                         "upstream gene variant",
                         "downstream gene variant",
                         "3 prime UTR variant",
                         "non coding transcript exon variant",
                         "5 prime UTR variant",
                         "splice",
                         "TF binding site variant",
                         "mature miRNA variant")

consequences.main.mean.snv.noncode <- sum_and_replace_rows(consequences.main.mean, consequences_levels)

# subset 
consequences.main.mean.snv.noncode <- subset(consequences.main.mean.snv.noncode, variant_type %in% consequences_levels)

# set levels
consequences.main.mean.snv.noncode$variant_type <- factor(consequences.main.mean.snv.noncode$variant_type, 
                                                       levels = consequences_levels)

consequences.main.mean.snv.noncode$cycle <- factor(consequences.main.mean.snv.noncode$cycle, 
                                                levels = c(1,3,6,9))

consequences.main.mean.snv.noncode$condition <- factor(consequences.main.mean.snv.noncode$condition, 
                                                    levels = c("control", "enu"))

# stats
for (variant in unique(consequences.main.mean.snv.noncode$variant_type)) {
  
  filtered_data <- consequences.main.mean.snv.noncode %>%
                         filter(variant_type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.mut ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.mut ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("consequences_cycle_main_snv_noncode_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- consequences.main.mean.snv.noncode %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE))

write.csv(stats_summary, "consequences_cycle_main_snv_noncode_stat_summary.csv")

# new facet labels
labels <- c("Intron",
            "Intergenic",
           "5KB Upstream",
           "5KB Downstream",
           "3' UTR",
           "lncRNA",
           "5' UTR",
           "Splice Region",
           "TF Binding Site",
           "miRNA")

names(labels) <- consequences_levels

# box plot
pdf("consequences_cycle_main_snv_noncode.pdf",
    width = 10,
    height = 5.5)
ggplot(consequences.main.mean.snv.noncode, aes(x = condition, y = obs.mut, fill = cycle)) + 
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, fill = cycle, y = mean, ymin = mean - sd, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", y = "Observed Variants Per Cell", fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    scale_y_continuous(trans=scales::pseudo_log_trans(base = 10), breaks = c(1, 10, 100, 1000, 10000)) +
    facet_wrap(.~variant_type,
               scale = "fixed",
               nrow = 2,
               labeller = labeller(variant_type = wrap_labels(labels, width = 10)))
dev.off()
```

##### INDEL variants
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/indel_coding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/consequences/cycle/indel_coding")

# subset 
consequences_levels <- c("inframe insertion",
                         "inframe deletion",
                         "frameshift variant")
consequences.main.mean.snv.code <- subset(consequences.main.mean, variant_type %in% consequences_levels)
consequences.main.mean.snv.code$variant_type <- factor(consequences.main.mean.snv.code$variant_type, levels = consequences_levels)

# stats
for (variant in unique(consequences.main.mean.snv.code$variant_type)) {
  
  filtered_data <- consequences.main.mean.snv.code %>%
                         filter(variant_type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.mut ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.mut ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("consequences_cycle_main_indel_code_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- consequences.main.mean.snv.code %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE))

write.csv(stats_summary, "consequences_cycle_main_indel_code_stat_summary.csv")

# new facet labels
labels <- c("Inframe insertion",
            "Inframe deletion",
            "Frameshift variant")
names(labels) <- consequences_levels

# box plot
pdf("consequences_cycle_main_indel_code.pdf",
    width = 6,
    height = 4)
ggplot(consequences.main.mean.snv.code, aes(x = condition, y = obs.mut, fill = cycle)) + 
    geom_errorbar(data = stats_summary, aes(x = condition, fill = cycle, y = mean, ymin = mean - 0.1, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
      geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", y = "Observed Variants Per Cell", fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    scale_y_continuous(limits = c(0, 1), breaks = c(0, 1)) +
    facet_wrap(.~variant_type,
               scale = "fixed",
               nrow = 1,
               labeller = labeller(variant_type = wrap_labels(labels, width = 15)))
dev.off()
```

## CADD

### By cycle

#### Observed variants
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd")

cadd.mean <- merge(cadd.mean,
                                    sample_table,
                                    by = "sample")


# remove undefined
cadd.mean <- subset(cadd.mean, !variant_type %in% c("undefined", "snv coding synonymous", "snv coding nonsynonymous", "snv noncoding intergenic"))

# set levels
levels <- c("snv noncoding intron",
            "snv noncoding transcript intron",
            "snv noncoding upstream gene",
            "snv noncoding downstream gene",
            "snv noncoding regulatory",
            "snv noncoding nmd",
            "snv noncoding 3prime UTR",
            "snv noncoding transcript exon",
            "snv noncoding splice",
            "snv noncoding 5prime UTR")

cadd.mean$variant_type <- factor(cadd.mean$variant_type,
                                       levels = levels)

for (variant in unique(cadd.mean$variant_type)) {
  
  filtered_data <- cadd.mean %>%
                         filter(variant_type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.mut ~ condition, data = filtered_data))
  
  # # Perform linear model
  # lm_result <- lm(obs.mut ~ condition * cycle, data = filtered_data)
  # 
  # # Compute emmeans and pairwise comparisons
  # emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  # pairwise_results <- pairs(emm_interaction)
  # 
  # # Adjust for multiple comparisons and add significance
  # adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
  #                     add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("cadd.mean_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  # write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = FALSE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- cadd.mean %>%
  group_by(condition, cycle, variant_type) %>%
  summarise(mean = mean(obs.mut, na.rm = TRUE),
            sd = sd(obs.mut, na.rm = TRUE)) %>%
  arrange(desc(mean))

write.csv(stats_summary, "cadd.mean_stat_summary.csv")

# # sort from greatest to least
# stats_summary_sort <- cadd.mean %>%
#   group_by(variant_type) %>%
#   summarise(mean = mean(obs.mut, na.rm = TRUE),
#             sd = sd(obs.mut, na.rm = TRUE)) %>%
#   arrange(desc(mean))
# 
# cadd.mean$variant_type <- factor(cadd.mean$variant_type,
#                          levels = stats_summary_sort$variant_type)

labels = c("Intron",
           "Non-coding RNA Intron",
           "5KB Upstream Gene",
           "5KB Downstream Gene",
           "Non-coding Regulatory",
           "Nonsense Mediated Decay",
           "3' UTR",
           "Non-coding RNA Exon",
           "Splice Site",
           "5' UTR")

names(labels) <- levels

pdf("cadd.mean.pdf",
    width = 10,
    height = 5)
ggplot(cadd.mean, aes(x = condition, y = obs.mut, fill = as.factor(cycle))) + 
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, fill = as.factor(cycle), y = mean, ymin = mean - sd, ymax = mean + sd),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", y = "Observed Variants Per Cell", fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        strip.text = element_text(size = 12)) +
    scale_x_discrete(labels = c("Control", "ENU")) +
    scale_y_continuous(trans=scales::pseudo_log_trans(base = 10), breaks = c(1, 10, 100, 1000, 10000)) +
    facet_wrap(.~variant_type,
               scale = "fixed",
               nrow = 2,
               labeller = labeller(variant_type = wrap_labels(labels, width = 15)))
dev.off()
```


#### SNV

##### Coding + Non-coding

###### Mean
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd")
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/snv")
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/snv/coding_noncoding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/snv/coding_noncoding")

cadd.mean$cycle <- factor(cadd.mean$cycle)

# stats
for (variant in unique(subset(cadd.mean, type == "snv coding" | type == "snv noncoding")$type)) {
  
  filtered_data <- cadd.mean %>%
                         filter(type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.score ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.score ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("cadd.mean_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# coding vs non coding stats
summary(aov(obs.score ~ type, data = cadd.mean[cadd.mean$type == "snv coding" | cadd.mean$type == "snv noncoding",]))

# get mean and sd for each group
stats_summary <- cadd.mean %>%
  group_by(condition, cycle, type) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.mean_stat_summary.csv")

labels <- c("Coding",
            "Non-coding")
names(labels) <- c("snv coding",
                   "snv noncoding")

# plot 
pdf("am_cycle_mean.pdf",
    width = 4,
    height = 4)
ggplot(subset(cadd.mean, type == "snv coding" | type == "snv noncoding"), aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
   # geom_errorbar(data = stats_summary, aes(x = condition, fill = cycle, y = mean, ymin = mean - 0.1, ymax = mean + sd),
   #                  width = 0.5, position = position_dodge(0.9)) +
   #    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Mean CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
  scale_y_log10() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~type,
                 scale = "fixed",
                 nrow = 1,
                 labeller = labeller(type = labels))
dev.off()
```

###### Median
```{r}
cadd.median$cycle <- factor(cadd.median$cycle)

# stats
for (variant in unique(subset(cadd.median, type == "snv coding" | type == "snv noncoding")$type)) {
  
  filtered_data <- cadd.median %>%
                         filter(type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.score ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.score ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("cadd.median_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- cadd.median %>%
  group_by(condition, cycle, type) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.median_stat_summary.csv")

labels <- c("Coding",
            "Non-coding")
names(labels) <- c("snv coding",
                   "snv noncoding")

# plot 
pdf("am_cycle_median.pdf",
    width = 5,
    height = 4)
ggplot(subset(cadd.median, type == "snv coding" | type == "snv noncoding"), aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Median CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~type,
                 scale = "fixed",
                 nrow = 1,
                 labeller = labeller(type = labels))
dev.off()
```

##### Non-coding

###### Mean
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/snv/noncoding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/snv/noncoding")

cadd.mean.noncoding <- subset(cadd.mean, type == "snv noncoding")

cadd.mean.noncoding$cycle <- factor(cadd.mean.noncoding$cycle)

# get mean and sd for each group
stats_summary <- cadd.mean.noncoding %>%
  group_by(condition, cycle) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.mean.noncoding_stat_summary.csv")

# plot 
pdf("am_cycle_mean.pdf",
    width = 3.5,
    height = 4)
ggplot(cadd.mean.noncoding, aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Mean CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU"))
dev.off()
```

###### Median
```{r}
cadd.median.noncoding <- subset(cadd.median, type == "snv noncoding")

cadd.median.noncoding$cycle <- factor(cadd.median.noncoding$cycle)

# get mean and sd for each group
stats_summary <- cadd.median.noncoding %>%
  group_by(condition, cycle) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.median.noncoding_stat_summary.csv")

# plot 
pdf("am_cycle_median.pdf",
    width = 3,
    height = 4)
ggplot(cadd.median.noncoding, aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Median CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU"))
dev.off()
```

#### INDEL

##### Coding + Non-coding

###### Mean
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd")
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/indel")
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/indel/coding_noncoding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/indel/coding_noncoding")

cadd.mean$cycle <- factor(cadd.mean$cycle)

# stats
for (variant in unique(cadd.mean$type)) {
  
  filtered_data <- cadd.mean %>%
                         filter(type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.score ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.score ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("cadd.mean_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- cadd.mean %>%
  group_by(condition, cycle, type) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.mean_stat_summary.csv")

labels <- c("Coding",
            "Non-coding")
names(labels) <- c("indel coding",
                   "indel noncoding")

# plot 
pdf("am_cycle_mean.pdf",
    width = 5,
    height = 4)
ggplot(subset(cadd.mean, type == "indel coding" | type == "indel noncoding"), aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Mean CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~type,
                 scale = "fixed",
                 nrow = 1,
                 labeller = labeller(type = labels))
dev.off()
```

###### Median
```{r}
cadd.median$cycle <- factor(cadd.median$cycle)

# stats
for (variant in unique(cadd.median$type)) {
  
  filtered_data <- cadd.median %>%
                         filter(type == variant)
  
  # Perform ANOVA
  anova_result <- summary(aov(obs.score ~ condition * cycle, data = filtered_data))
  
  # Perform linear model
  lm_result <- lm(obs.score ~ condition * cycle, data = filtered_data)

  # Compute emmeans and pairwise comparisons
  emm_interaction <- emmeans(lm_result, ~ condition * cycle)
  pairwise_results <- pairs(emm_interaction)

  # Adjust for multiple comparisons and add significance
  adjusted_results <- na.omit(summary(pairwise_results, adjust = "tukey")) %>%
                      add_significance("p.value")

  # Construct the output filename
  output_filename <- paste("cadd.median_stats_", 
                            gsub(" ", "_", variant), ".csv", sep = "")

  # Write the results to a CSV file
  write.csv(adjusted_results, output_filename)
  anova_summary_text <- write(capture.output(anova_result), 
                              file = output_filename, 
                              append = TRUE, 
                              sep = "\n")
  
}

# get mean and sd for each group
stats_summary <- cadd.median %>%
  group_by(condition, cycle, type) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.median_stat_summary.csv")

labels <- c("Coding",
            "Non-coding")
names(labels) <- c("indel coding",
                   "indel noncoding")

# plot 
pdf("am_cycle_median.pdf",
    width = 5,
    height = 4)
ggplot(subset(cadd.median, type == "indel coding" | type == "indel noncoding"), aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_hline(yintercept = 0, linetype = "dashed") +
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Median CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU")) +
    facet_wrap(.~type,
                 scale = "fixed",
                 nrow = 1,
                 labeller = labeller(type = labels))
dev.off()
```

##### Non-coding

###### Mean
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/indel/noncoding")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/5-variant-effect-prediction/3-number-plot/cadd/indel/noncoding")

cadd.mean.noncoding <- subset(cadd.mean, type == "indel noncoding")

cadd.mean.noncoding$cycle <- factor(cadd.mean.noncoding$cycle)

# get mean and sd for each group
stats_summary <- cadd.mean.noncoding %>%
  group_by(condition, cycle) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.mean.noncoding_stat_summary.csv")

# plot 
pdf("am_cycle_mean.pdf",
    width = 3,
    height = 4)
ggplot(cadd.mean.noncoding, aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Mean CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU"))
dev.off()
```

###### Median
```{r}
cadd.median.noncoding <- subset(cadd.median, type == "indel noncoding")

cadd.median.noncoding$cycle <- factor(cadd.median.noncoding$cycle)

# get mean and sd for each group
stats_summary <- cadd.median.noncoding %>%
  group_by(condition, cycle) %>%
  summarise(mean = mean(obs.score, na.rm = TRUE),
            sd = sd(obs.score, na.rm = TRUE))

write.csv(stats_summary, "cadd.median.noncoding_stat_summary.csv")

# plot 
pdf("am_cycle_median.pdf",
    width = 3,
    height = 4)
ggplot(cadd.median.noncoding, aes(x = condition, y = obs.score, fill = cycle)) + 
    geom_boxplot(outlier.shape = NA) +
    geom_point(size = 2, alpha = 0.5, position = position_dodge(width = 0.75)) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "", 
         y = "Median CADD Score Per Cell", 
         fill = "Cycle") +
    scale_fill_viridis(discrete = TRUE, end = 0.8) +
    theme_Publication() +
    theme(legend.position = "top",  # Place legend on the right side
        legend.box = "horizontal",  # Stack legend items vertically
        legend.text = element_text(size = 10),  # Adjust legend text size
        legend.title = element_text(size = 12)) +  # Adjust legend title size
    scale_x_discrete(labels = c("Control", "ENU"))
dev.off()
```

## Impact
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/impact")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/impact")

# remove duplicates
table(vep_list[[21]]$IMPACT)
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- subset(df, "IMPACT" != "-")
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})

# count the different types of consequences in each sample
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
    group_by(IMPACT) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -IMPACT)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# remove modifier
consequences_df_long <- subset(consequences_df_long, IMPACT != "MODIFIER")

# set order
consequences_df_long$IMPACT <- factor(consequences_df_long$IMPACT, levels = c("LOW", "MODERATE", "HIGH"))

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different categories
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_consequence = Count/sum(Count) * 100)

##cycle
# plot observed
pdf("snv_impact_observed_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot estimated
pdf("snv_impact_estimated_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot percentages
pdf("snv_impact_percent_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

# percentages of consequences
pdf("snv_impact_percent_type_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = percentage_consequence, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% IMPACT type Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

##condition bar plot
# plot observed
stats_summary <- consequences_df_long %>%
  group_by(condition, IMPACT) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_impact_observed_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, IMPACT) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_impact_estimated_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, IMPACT) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_impact_percent_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- consequences_df_long %>%
  group_by(condition, IMPACT) %>%
  summarise(mean_log2fc = mean(percentage_consequence, na.rm = TRUE),
            sd_log2fc = sd(percentage_consequence, na.rm = TRUE))

pdf("snv_impact_percent_type_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = percentage_consequence, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = percentage_consequence), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% IMPACT type Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~IMPACT, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

## CAROL
- For missense variants, combination of SIFT and PolyPhen scores
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/carol")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/carol")

# remove duplicates
table(vep_list[[21]]$CAROL)
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- subset(df, CAROL != "-")
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
table(vep_list_no_dup[[21]]$CAROL)

# count the different types of consequences in each sample
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
    mutate(CAROL_group = sub("\\(.*", "", CAROL)) %>%
    group_by(CAROL_group) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% spread(key = Sample, value = Count, fill = 0)
consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -CAROL_group)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# set order
consequences_df_long$CAROL_group <- factor(consequences_df_long$CAROL_group, levels = c("Neutral", "Deleterious"))

# calculate percentages
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# percentages of different categories
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_consequence = Count/sum(Count) * 100)

##cycle
# plot observed
pdf("snv_carol_observed_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) + 
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot estimated
pdf("snv_carol_estimated_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot percentages
pdf("snv_carol_percent_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

# percentages of consequences
pdf("snv_carol_percent_type_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = percentage_consequence, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% CAROL type") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

##condition bar plot
# plot observed
stats_summary <- consequences_df_long %>%
  group_by(condition, CAROL_group) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_carol_observed_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, CAROL_group) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_carol_estimated_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, CAROL_group) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_carol_percent_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- consequences_df_long %>%
  group_by(condition, CAROL_group) %>%
  summarise(mean_log2fc = mean(percentage_consequence, na.rm = TRUE),
            sd_log2fc = sd(percentage_consequence, na.rm = TRUE))

pdf("snv_carol_percent_type_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = percentage_consequence, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = percentage_consequence), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% CAROL type") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~CAROL_group, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

## BLOSUM62

### Categorize scores
- >0 is conservative, <0 is non-conservative
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/blosum62")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/blosum62")

# remove duplicates
summary(as.numeric(vep_list[[21]]$BLOSUM62))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$BLOSUM62)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$BLOSUM62))


pdf("blosum62_distribution.pdf",
    width = 5,
    height = 5)
hist(as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["BLOSUM62"]]))),
     main = NULL,
     xlab = "BLOSUM62")
dev.off()

# count the different types of scores in each sample
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
    group_by(BLOSUM62) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -BLOSUM62)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$BLOSUM62 <- factor(consequences_df_long$BLOSUM62, levels = seq(-3, 3, 1))

##cycle
# plot observed
pdf("snv_BLOSUM62_observed_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot estimated
pdf("snv_BLOSUM62_estimated_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot percentages
pdf("snv_BLOSUM62_percent_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

# percentages of consequences
pdf("snv_BLOSUM62_percent_type_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% BLOSUM62 type Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

##condition bar plot
# plot observed
stats_summary <- consequences_df_long %>%
  group_by(condition, BLOSUM62) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_BLOSUM62_observed_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, BLOSUM62) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_BLOSUM62_estimated_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, BLOSUM62) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_BLOSUM62_percent_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- consequences_df_long %>%
  group_by(condition, BLOSUM62) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_BLOSUM62_percent_type_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% BLOSUM62 type Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~BLOSUM62, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```
### Mean/Median scores
- >0 is conservative, <0 is non-conservative
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/blosum62")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/blosum62")

# remove duplicates
summary(as.numeric(vep_list[[21]]$BLOSUM62))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$BLOSUM62)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$BLOSUM62))


pdf("blosum62_distribution.pdf",
    width = 5,
    height = 5)
hist(as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["BLOSUM62"]]))),
     main = NULL,
     xlab = "BLOSUM62")
dev.off()

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  data.frame(
    Sum = sum(as.numeric(df$BLOSUM62), na.rm = TRUE),
    Mean = mean(as.numeric(df$BLOSUM62), na.rm = TRUE),
    Median = median(as.numeric(df$BLOSUM62), na.rm = TRUE),
    SD = sd(as.numeric(df$BLOSUM62), na.rm = TRUE)
  )
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results) 
consequences_df_long <- cbind(consequences_df_long, sample_table) # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_blosum62_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean BLOSUM62 Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

pdf("snv_blosum62_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median BLOSUM62 Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

##condition bar plot

# mean
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_blosum62_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean BLOSUM62 Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_blosum62_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median BLOSUM62 Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()
```


## Alpha Missense
### Class
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/alpha-missense")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/alpha-missense")

# remove duplicates
table(vep_list[[21]]$am_class)
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- subset(df, am_class != "-")
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
table(vep_list_no_dup[[21]]$am_class)

# count the different types of consequences in each sample
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
    group_by(am_class) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[-1,]

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -am_class)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info
# set order
consequences_df_long$am_class <- factor(consequences_df_long$am_class, levels = c("ambiguous", "benign", "pathogenic"))

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different categories
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_consequence = Count/sum(Count) * 100)

##cycle
# plot observed
pdf("snv_am_class_observed_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot estimated
pdf("snv_am_class_estimated_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot percentages
pdf("snv_am_class_percent_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

# percentages of consequences
pdf("snv_am_class_percent_type_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = percentage_consequence, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% Class type") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

##condition bar plot
# plot observed
stats_summary <- consequences_df_long %>%
  group_by(condition, am_class) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_am_class_observed_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, am_class) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_am_class_estimated_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, am_class) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_am_class_percent_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- consequences_df_long %>%
  group_by(condition, am_class) %>%
  summarise(mean_log2fc = mean(percentage_consequence, na.rm = TRUE),
            sd_log2fc = sd(percentage_consequence, na.rm = TRUE))

pdf("snv_am_class_percent_type_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = percentage_consequence, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = percentage_consequence), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Class type") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~am_class, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

### Pathogenicity score
- 'Likely benign' if am_pathogenicity < 0.34
- 'Likely pathogenic' if am_pathogenicity > 0.564
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/alpha-missense")

# remove duplicates
summary(as.numeric(vep_list[[21]]$am_pathogenicity))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$am_pathogenicity)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$am_pathogenicity))

pdf("am_pathogenicity_distribution.pdf",
    width = 5,
    height = 5)
hist(as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["am_pathogenicity"]]))),
     main = NULL,
     xlab = "Pathogenicity")
dev.off()

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  data.frame(
    Sum = sum(as.numeric(df$CADD_RAW), na.rm = TRUE),
    Mean = mean(as.numeric(df$CADD_RAW), na.rm = TRUE),
    Median = median(as.numeric(df$CADD_RAW), na.rm = TRUE),
    SD = sd(as.numeric(df$CADD_RAW), na.rm = TRUE)
  )
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results) 
consequences_df_long <- cbind(consequences_df_long, sample_table) # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_am_pathogenicity_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Pathogenicity Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

pdf("snv_am_pathogenicity_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Pathogenicity Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_am_pathogenicity_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Pathogenicity Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_am_pathogenicity_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Pathogenicity Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()
```

## CADD
- coding and non-coding
- scoring the deleteriousness of single nucleotide variants
- higher the raw C score the more predicted to be deleterious 

### Categorize scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/cadd")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/cadd")

table(vep_list[[21]]$Consequence)
# remove duplicates
summary(as.numeric(vep_list[[21]]$CADD_RAW))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$CADD_RAW)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["CADD_RAW"]])))
summary(scores)

pdf("RAW_CADD_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "Raw CADD")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- seq(min(scores), max(scores), (max(scores) - min(scores))/10)
boundaries[1] <- min(scores)
boundaries[length(boundaries)] <- max(scores)

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(CADD_RAW), boundaries, labels = paste0(seq(10, 100, by = 10), "th percentile"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = unique(consequences_df_long$buckets))

##cycle
# plot observed
pdf("snv_CADD_categ_observed_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_CADD_categ_observed_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_CADD_categ_estimated_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_CADD_categ_estimated_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_CADD_categ_percent_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_CADD_categ_percent_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_CADD_categ_percent_type_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_CADD_categ_percent_type_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_CADD_categ_observed_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_CADD_categ_observed_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_CADD_categ_estimated_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_CADD_categ_estimated_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_CADD_categ_percentage_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_CADD_categ_percentage_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_CADD_categ_percentage_type_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_CADD_categ_percentage_type_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

### Mean/Median
- higher the raw C score the more predicted to be deleterious 
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/cadd")

# remove duplicates
summary(as.numeric(vep_list[[21]]$CADD_RAW))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$CADD_RAW)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["CADD_RAW"]]))))

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "Coding",
    TRUE ~ "Non-Coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
  group_by(type) %>%
    summarise(Sum = sum(as.numeric(CADD_RAW), na.rm = TRUE),
              Mean = mean(as.numeric(CADD_RAW), na.rm = TRUE),
              Median = median(as.numeric(CADD_RAW), na.rm = TRUE),
              SD = sd(as.numeric(CADD_RAW), na.rm = TRUE))
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results)
consequences_df_long$sample <- rep(sample_table$sample, each = 2)
consequences_df_long <- merge(consequences_df_long, sample_table, by = "sample") # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_cadd_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Raw CADD Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

pdf("snv_cadd_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Raw CADD Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)  +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_cadd_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Raw CADD Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)   +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_cadd_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Raw CADD Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)    +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()
```

## TF Motif
### Plot change in high info position
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/tf-motif")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/tf-motif")

# remove duplicates
table(vep_list[[21]]$HIGH_INF_POS)
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- subset(df, HIGH_INF_POS != "-")
  #df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})

# count the different types of consequences in each sample
# TODO: only look at motifs in promoters or enhancers
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
    
    group_by(HIGH_INF_POS) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df$HIGH_INF_POS <- c("Low information", "High information")

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -HIGH_INF_POS)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# set order
consequences_df_long$HIGH_INF_POS <- factor(consequences_df_long$HIGH_INF_POS, levels = c("Low information", "High information"))

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different categories
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_consequence = Count/sum(Count) * 100)

##cycle
# plot observed
pdf("snv_motif_info_observed_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot estimated
pdf("snv_motif_info_estimated_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
  scale_y_log10()
dev.off()

# plot percentages
pdf("snv_motif_info_percent_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

# percentages of consequences
pdf("snv_motif_info_percent_type_cycle_all.pdf",
    width = 6,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = percentage_consequence, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% motif_info type") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40))
dev.off()

##condition bar plot
# plot observed
stats_summary <- consequences_df_long %>%
  group_by(condition, HIGH_INF_POS) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_motif_info_observed_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, HIGH_INF_POS) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_motif_info_estimated_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- consequences_df_long %>%
  group_by(condition, HIGH_INF_POS) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_am_class_percent_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- consequences_df_long %>%
  group_by(condition, HIGH_INF_POS) %>%
  summarise(mean_log2fc = mean(percentage_consequence, na.rm = TRUE),
            sd_log2fc = sd(percentage_consequence, na.rm = TRUE))

pdf("snv_am_class_percent_type_condition_all.pdf",
    width = 5,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = percentage_consequence, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = percentage_consequence), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Motif Info type") +
    scale_y_continuous(labels = function(x) paste0(x, "%")) +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_grid(.~HIGH_INF_POS, scale = "fixed", labeller = label_wrap_gen(width = 40)) +
    theme(strip.text = element_text(size = 12),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

### Plot Z-score
- unsure why mean Z-scores are so low, using median
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/tf-motif")

# remove duplicates
summary(as.numeric(vep_list[[21]]$MOTIF_SCORE_CHANGE))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$MOTIF_SCORE_CHANGE)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$MOTIF_SCORE_CHANGE))

score <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["MOTIF_SCORE_CHANGE"]])))

pdf("motif_z-score_distribution.pdf",
    width = 5,
    height = 5)
hist(score,
     main = NULL,
     xlab = "MOTIF_SCORE_CHANGE")
dev.off()

pdf("motif_z-score_distribution.pdf",
    width = 5,
    height = 5)
hist((score - mean(score))/sd(score),
     main = NULL,
     xlab = "Motif Z-score")
dev.off()

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  df %>% mutate(zscore = (as.numeric(MOTIF_SCORE_CHANGE) - mean(as.numeric(MOTIF_SCORE_CHANGE)))/sd(as.numeric(MOTIF_SCORE_CHANGE))) %>%
    summarise(Sum = sum(abs((zscore)), na.rm = TRUE),
              Mean = mean(zscore),
              Median = median((zscore), na.rm = TRUE),
              SD = sd((zscore), na.rm = TRUE))
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results) 
consequences_df_long <- cbind(consequences_df_long, sample_table) # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_motif_z-score_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Motif Z-score Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

pdf("snv_motif_z-score_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Motif Z-score Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

pdf("snv_motif_z-score_sum_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Sum, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Cumulative Motif Z-score Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_motif_z-score_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Motif Z-score Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_motif_z-score_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Motif Z-score Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()

# sum 
stats_summary <- consequences_df_long %>%
  group_by(condition) %>%
  summarise(mean_log2fc = mean(Sum, na.rm = TRUE),
            sd_log2fc = sd(Sum, na.rm = TRUE))

pdf("snv_motif_z-score_sum_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Sum, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Sum), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Sum Motif Z-score Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)
dev.off()
```



## Conservation

### GERP
#### Categorize scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")

# remove duplicates
summary(as.numeric(vep_list[[21]]$GERP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$GERP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["GERP"]])))
summary(scores)

pdf("GERP_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "GERP")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- c(min(scores), 0, max(scores))


# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(GERP), boundaries, labels = c("Neutral", "Conserved"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = c("Neutral", "Conserved"))

##cycle
# plot observed
pdf("snv_GERP_categ_observed_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_categ_observed_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_GERP_categ_estimated_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_categ_estimated_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_GERP_categ_percent_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_categ_percent_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_GERP_categ_percent_type_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_categ_percent_type_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_GERP_categ_observed_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_GERP_categ_observed_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_GERP_categ_estimated_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_GERP_categ_estimated_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_GERP_categ_percentage_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_GERP_categ_percentage_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_GERP_categ_percentage_type_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_GERP_categ_percentage_type_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Percentile scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")

# remove duplicates
summary(as.numeric(vep_list[[21]]$GERP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$GERP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["GERP"]])))
summary(scores)

pdf("GERP_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "GERP")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- seq(min(scores), max(scores), (max(scores) - min(scores))/10)
boundaries[1] <- min(scores)
boundaries[length(boundaries)] <- max(scores)

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(GERP), boundaries, labels = paste0(seq(10, 100, by = 10), "th percentile"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = unique(consequences_df_long$buckets))

##cycle
# plot observed
pdf("snv_GERP_percentile_observed_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_percentile_observed_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_GERP_percentile_estimated_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_percentile_estimated_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10()  +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_GERP_percentile_percent_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_percentile_percent_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_GERP_percentile_percent_type_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_GERP_percentile_percent_type_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_GERP_percentile_observed_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_GERP_percentile_observed_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_GERP_percentile_estimated_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_GERP_percentile_estimated_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_GERP_percentile_percentage_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_GERP_percentile_percentage_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_GERP_percentile_percentage_type_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_GERP_percentile_percentage_type_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Mean/Median score
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/gerp")

# remove duplicates
summary(as.numeric(vep_list[[21]]$GERP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$GERP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$GERP))

pdf("gerp_score_distribution.pdf",
    width = 5,
    height = 5)
hist(as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["GERP"]]))),
     main = NULL,
     xlab = "GERP")
dev.off()

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  # Add a filter step to only include rows where GERP score is greater than 0, because negatiive scores are not interpretable
  filter(as.numeric(GERP) > 0) %>%
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "Coding",
    TRUE ~ "Non-Coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
  group_by(type) %>%
    summarise(Sum = sum(as.numeric(GERP), na.rm = TRUE),
              Mean = mean(as.numeric(GERP), na.rm = TRUE),
              Median = median(as.numeric(GERP), na.rm = TRUE),
              SD = sd(as.numeric(GERP), na.rm = TRUE))
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results)
consequences_df_long$sample <- rep(sample_table$sample, each = 2)
consequences_df_long <- merge(consequences_df_long, sample_table, by = "sample") # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_GERP_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Raw GERP Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

pdf("snv_GERP_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Raw GERP Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)  +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_GERP_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Raw GERP Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)   +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_GERP_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Raw GERP Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)    +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()
```

### PhyloP
#### Categorize scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phylop")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phylop")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phyloP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phyloP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["phyloP"]])))
summary(scores)

pdf("phyloP_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "phyloP")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- c(min(scores), -1.3, 1.3, max(scores))

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(phyloP), boundaries, labels = c("Non-conserved", "Neutral", "Conserved"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = c("Non-conserved", "Neutral", "Conserved"))

##cycle
# plot observed
pdf("snv_phyloP_categ_observed_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_categ_observed_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_phyloP_categ_estimated_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_categ_estimated_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_phyloP_categ_percent_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_categ_percent_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_phyloP_categ_percent_type_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_categ_percent_type_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phyloP_categ_observed_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phyloP_categ_observed_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phyloP_categ_estimated_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phyloP_categ_estimated_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phyloP_categ_percentage_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phyloP_categ_percentage_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phyloP_categ_percentage_type_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phyloP_categ_percentage_type_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Percentile scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phylop")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phylop")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phyloP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phyloP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["phyloP"]])))
summary(scores)s

pdf("phyloP_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "phyloP")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- seq(min(scores), max(scores), (max(scores) - min(scores))/10)
boundaries[1] <- min(scores)
boundaries[length(boundaries)] <- max(scores)

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(phyloP), boundaries, labels = paste0(seq(10, 100, by = 10), "th percentile"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = paste0(seq(10, 100, by = 10), "th percentile"))

##cycle
# plot observed
pdf("snv_phyloP_percentile_observed_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_percentile_observed_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_phyloP_percentile_estimated_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_percentile_estimated_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_phyloP_percentile_percent_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_percentile_percent_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_phyloP_percentile_percent_type_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phyloP_percentile_percent_type_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phyloP_percentile_observed_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phyloP_percentile_observed_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phyloP_percentile_estimated_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phyloP_percentile_estimated_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phyloP_percentile_percentage_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phyloP_percentile_percentage_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phyloP_percentile_percentage_type_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phyloP_percentile_percentage_type_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Mean/Median score
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phylop")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phyloP))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phyloP)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$phyloP))

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "Coding",
    TRUE ~ "Non-Coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
  group_by(type) %>%
    summarise(Sum = sum(as.numeric(phyloP), na.rm = TRUE),
              Mean = mean(as.numeric(phyloP), na.rm = TRUE),
              Median = median(as.numeric(phyloP), na.rm = TRUE),
              SD = sd(as.numeric(phyloP), na.rm = TRUE))
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results)
consequences_df_long$sample <- rep(sample_table$sample, each = 2)
consequences_df_long <- merge(consequences_df_long, sample_table, by = "sample") # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_phyloP_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Raw phyloP Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

pdf("snv_phyloP_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Raw phyloP Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)  +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_phyloP_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Raw phyloP Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)   +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_phyloP_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Raw phyloP Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)    +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()
```

### PhastCons score
#### Categorize scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phastcons")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phastcons")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phastCons))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phastCons)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["phastCons"]])))
summary(scores)

pdf("phastCons_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "phastCons")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- c(0,0.5,1)
boundaries[1] <- min(scores)
boundaries[length(boundaries)] <- max(scores)

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(phastCons), boundaries, labels = c("Non-conserved", "Conserved"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = c("Non-conserved", "Conserved"))

##cycle
# plot observed
pdf("snv_phastCons_categ_observed_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_categ_observed_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_phastCons_categ_estimated_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_categ_estimated_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_phastCons_categ_percent_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_categ_percent_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_phastCons_categ_percent_type_cycle_all_coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_categ_percent_type_cycle_all_non-coding.pdf",
    width = 5,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phastCons_categ_observed_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phastCons_categ_observed_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phastCons_categ_estimated_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phastCons_categ_estimated_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 1, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phastCons_categ_percentage_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phastCons_categ_percentage_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phastCons_categ_percentage_type_condition_all_coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phastCons_categ_percentage_type_condition_all_non-coding.pdf",
    width = 5,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 1, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Percentile scores
- break up by coding and non-coding
```{r}
dir.create("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phastcons")
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phastcons")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phastCons))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phastCons)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
scores <- as.numeric(unlist(lapply(vep_list_no_dup, function(x) x[["phastCons"]])))
summary(scores)

pdf("phastCons_distribution.pdf",
    width = 5,
    height = 5)
hist(scores,
     main = NULL,
     xlab = "phastCons")
dev.off()

# break scores into 10 buckets - make sure min and max are set correctly
boundaries <- seq(min(scores), max(scores), (max(scores) - min(scores))/10)
boundaries[1] <- min(scores)
boundaries[length(boundaries)] <- max(scores)

# count the occurences of scores in each bucket
vep_list_agg <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "coding",
    TRUE ~ "non-coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
    mutate(buckets = cut(as.numeric(phastCons), boundaries, labels = paste0(seq(10, 100, by = 10), "th percentile"))) %>% 
    group_by(type, buckets) %>% 
    summarise(Count = n())
})

# add sample identifiers
vep_list_agg <- mapply(function(df, name) {
  df$Sample <- name
  return(df)
}, vep_list_agg, sample_table$sample, SIMPLIFY = FALSE)

# merge into df
consequences_df <- do.call(rbind, vep_list_agg)
consequences_df <- consequences_df %>% 
  spread(key = Sample, value = Count, fill = 0)
# consequences_df <- consequences_df[!rowSums(consequences_df[,-1]) == 0,] # remove rows with 0
consequences_df <- consequences_df[!is.na(consequences_df$buckets),] # remove bucket NA rows

# make it long
consequences_df_long <- consequences_df %>% gather(Sample, Count, -type, -buckets)
consequences_df_long <- merge(consequences_df_long, sample_table, by.x = "Sample", by.y = "sample") # add sample info

# calculate estimated mutations
consequences_df_long$SNV_est <- consequences_df_long$Count / consequences_df_long$Cov / consequences_df_long$Sen_SNV * hg19_size

# calculate percentages of snvs in each consequence
consequences_df_long$Percentage <- (consequences_df_long$Count / consequences_df_long$SNV_obs) * 100

# percentages of different consequences
consequences_df_long <- consequences_df_long %>%
  group_by(Sample) %>%
  mutate(percentage_type = Count/sum(Count) * 100)

# order
consequences_df_long$type <- factor(consequences_df_long$type, levels = unique(consequences_df_long$type))
consequences_df_long$buckets <- factor(consequences_df_long$buckets, levels = paste0(seq(10, 100, by = 10), "th percentile"))

##cycle
# plot observed
pdf("snv_phastCons_percentile_observed_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_percentile_observed_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Count, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Observed SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot estimated
pdf("snv_phastCons_percentile_estimated_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_percentile_estimated_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = SNV_est, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
  theme(strip.text = element_text(size = 7))
dev.off()

# plot percentages
pdf("snv_phastCons_percentile_percent_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_percentile_percent_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = Percentage, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% SNVs Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

# percentages of consequences
pdf("snv_phastCons_percentile_percent_type_cycle_all_coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

pdf("snv_phastCons_percentile_percent_type_cycle_all_non-coding.pdf",
    width = 8,
    height = 5)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = factor(cycle), y = percentage_type, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "% buckets type Per Cell", title = "Non-Coding") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free", labeller = label_wrap_gen(width = 40))  +
  theme(strip.text = element_text(size = 7))
dev.off()

##condition bar plot
# plot observed
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phastCons_percentile_observed_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Count, na.rm = TRUE),
            sd_log2fc = sd(Count, na.rm = TRUE))

pdf("snv_phastCons_percentile_observed_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Count, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Count), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Observed SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# estimated
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phastCons_percentile_estimated_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(SNV_est, na.rm = TRUE),
            sd_log2fc = sd(SNV_est, na.rm = TRUE))

pdf("snv_phastCons_percentile_estimated_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = SNV_est, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = SNV_est), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "Estimated SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
  facet_wrap(.~buckets, nrow = 2, scale = "fixed") +
  scale_y_log10() +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percentage
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phastCons_percentile_percentage_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(Percentage, na.rm = TRUE),
            sd_log2fc = sd(Percentage, na.rm = TRUE))

pdf("snv_phastCons_percentile_percentage_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = Percentage, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = Percentage), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% SNVs Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

# percent of type
stats_summary <- subset(consequences_df_long, type == "coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phastCons_percentile_percentage_type_condition_all_coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()

stats_summary <- subset(consequences_df_long, type == "non-coding") %>%
  group_by(condition, buckets) %>%
  summarise(mean_log2fc = mean(percentage_type, na.rm = TRUE),
            sd_log2fc = sd(percentage_type, na.rm = TRUE))

pdf("snv_phastCons_percentile_percentage_type_condition_all_non-coding.pdf",
    width = 8,
    height = 4)
ggplot(subset(consequences_df_long, type == "non-coding"), aes(x = condition, y = percentage_type, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
      geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(aes(x = condition, y = percentage_type), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
    labs(x = "Group", y = "% Buckets type Per Cell", title = "Non-Coding") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16) +
      scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(.~buckets, nrow = 2, scale = "free") +
    theme(strip.text = element_text(size = 7),
          axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank())
dev.off()
```

#### Mean/Median score
```{r}
setwd("/Users/ronaldcutler/Dropbox\ (EinsteinMed)/Vijg-lab/Projects/mutation\ accumulation/231009\ multiple\ ENU\ analysis/6-variant-effect-prediction/phastCons")

# remove duplicates
summary(as.numeric(vep_list[[21]]$phastCons))
vep_list_no_dup <- lapply(vep_list, function(df) {
  df <- df[!is.na(as.numeric(df$phastCons)),]
  df <- distinct(df, Uploaded_variation, .keep_all = TRUE)
  return(df)
})
summary(as.numeric(vep_list_no_dup[[21]]$phastCons))

# get mean/median Z-score per cell
results <- lapply(vep_list_no_dup, function(df) {
  df %>% 
  mutate(type = case_when(
    grepl("synonymous|missense_variant|stop_gained|start_lost|stop_retained", Consequence) ~ "Coding",
    TRUE ~ "Non-Coding" # The catch-all condition to label everything else as "non-coding"
  )) %>%
  group_by(type) %>%
    summarise(Sum = sum(as.numeric(phastCons), na.rm = TRUE),
              Mean = mean(as.numeric(phastCons), na.rm = TRUE),
              Median = median(as.numeric(phastCons), na.rm = TRUE),
              SD = sd(as.numeric(phastCons), na.rm = TRUE))
})

# Combine the results into a new dataframe
consequences_df_long <- do.call(rbind, results)
consequences_df_long$sample <- rep(sample_table$sample, each = 2)
consequences_df_long <- merge(consequences_df_long, sample_table, by = "sample") # add sample info assuming order is the same as in sample table

##cycle
pdf("snv_phastCons_mean_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Mean, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Mean Raw phastCons Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16) +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

pdf("snv_phastCons_median_cycle_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = factor(cycle), y = Median, fill = condition)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
  guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Cycle", y = "Median Raw phastCons Per Cell") +
  scale_fill_Publication() +
  theme_Publication(base_size=16)  +
  facet_wrap(.~type, nrow = 1, scale = "free")
dev.off()

##condition bar plot
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Mean, na.rm = TRUE),
            sd_log2fc = sd(Mean, na.rm = TRUE))

pdf("snv_phastCons_mean_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Mean, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Mean), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Mean Raw phastCons Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)   +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()

# median 
stats_summary <- consequences_df_long %>%
  group_by(condition, type) %>%
  summarise(mean_log2fc = mean(Median, na.rm = TRUE),
            sd_log2fc = sd(Median, na.rm = TRUE))

pdf("snv_phastCons_median_condition_all.pdf",
    width = 4,
    height = 4)
ggplot(consequences_df_long, aes(x = condition, y = Median, fill = condition)) +
    geom_bar(stat = "summary", fun.y = "mean", position = "dodge") +  # Average bar plot
    geom_errorbar(data = stats_summary, aes(x = condition, y = mean_log2fc, ymin = mean_log2fc - sd_log2fc, ymax = mean_log2fc + sd_log2fc),
                    width = 0.5, position = position_dodge(0.9)) +
    geom_point(data = consequences_df_long, aes(x = condition, y = Median), position = position_jitterdodge(jitter.width = 0.2), size = 2, alpha = 0.5) +
    stat_compare_means(method = "t.test", label = "p.signif",  vjust = 1, label.x.npc = "center", hide.ns = TRUE, size = 5) +
    guides(fill = guide_legend(override.aes = list(shape = NA))) +
  labs(x = "Group", y = "Median Raw phastCons Per Cell") +
    scale_fill_Publication() +
    theme_Publication(base_size = 16)    +
  facet_wrap(.~type, nrow = 1, scale = "free") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank())
dev.off()
```
